src/Standards/Generic/Sniffs/PHP/DisallowRequestSuperGlobalSniff.php

```diff
@@ -0,0 +1,55 @@
+<?php
+/**
+ * Ensures the $_REQUEST super global is not used
+ *
+ * @author    Jeantwan Teuma <jeant.m24@gmail.com>
+ * @copyright 2006-2019 Squiz Pty Ltd (ABN 77 084 670 600)
+ * @license   https://github.com/squizlabs/PHP_CodeSniffer/blob/master/licence.txt BSD Licence
+ */
+
+namespace PHP_CodeSniffer\Standards\Generic\Sniffs\PHP;
+
+use PHP_CodeSniffer\Files\File;
+use PHP_CodeSniffer\Sniffs\Sniff;
+
+class DisallowRequestSuperGlobalSniff implements Sniff
+{
+
+
+    /**
+     * Returns an array of tokens this test wants to listen for.
+     *
+     * @return array
+     */
+    public function register()
+    {
+        return [T_VARIABLE];
+
+    }//end register()
+
+
+    /**
+     * Processes this sniff, when one of its tokens is encountered.
+     *
+     * @param File $phpcsFile The file being scanned.
+     * @param int  $stackPtr  The position of the current token in the stack passed in $tokens.
+     *
+     * @return void
+     */
+    public function process(File $phpcsFile, $stackPtr)
+    {
+        $tokens = $phpcsFile->getTokens();
+
+        $varName = $tokens[$stackPtr]['content'];
+        if ($varName !== '$_REQUEST') {
+            return;
+        }
+
+        $type  = 'RequestSuperGlobalAccessed';
+        $error = 'The $_REQUEST super global should not be used. Use $_GET, $_POST or $_COOKIE instead';
+        $phpcsFile->addError($error, $stackPtr, $type, []);
```

Comment:
What I would normally do here is use `Found` as the error type string. This typically allows for cleaner message codes. For example, `Generic.PHP.DisallowRequestSuperGlobal.Found` instead of `Generic.PHP.DisallowRequestSuperGlobal.RequestSuperGlobalAccessed`.

I don't think it needs a variable defined for it the type either. And no need to pass an empty array as the 4th argument as this is the default value already.

Reply:
Suggestions implemented